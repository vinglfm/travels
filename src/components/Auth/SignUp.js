import React, {Component} from 'react';
import PropTypes from 'prop-types';
import Validator from 'fastest-validator';
import styles from './AuthModal.css';

const validator = new Validator();

export default class SignUp extends Component {
    constructor(props) {
        super(props);

        this.state = {
            email:'',
            password:'',
            firstName: '',
            lastName: ''
        };

        this.emailValidationSchema = {
            email: { type: 'email' }
        };

        this.passwordValidationSchema = {
            password: { type: 'string', min: 5, max: 12 }
        }

        this.signUp = this.signUp.bind(this);
        this.validateEmail = this.validateEmail.bind(this);
        this.validatePassword = this.validatePassword.bind(this);
        this.handleChange = this.handleChange.bind(this);
    }

    signUp() {
        console.log(this.state);
        this.props.onSignUp(this.state);
    }

    handleChange(elem) {
        this.setState({[elem.target.name]: elem.target.value});
        console.log(elem.target.name, '=', elem.target.value);
    }

    validateEmail() {
        return validator.validate(this.state, this.emailValidationSchema);
    }

    validatePassword() {
        return validator.validate(this.state, this.passwordValidationSchema);
    }

    render() {
        const emailValidationError = this.validateEmail();
        const passwordValidationError = this.validatePassword();
        const emailErrorMessage = emailValidationError[0] ? emailValidationError[0].message: '';
        const passwordErrorMessage = passwordValidationError[0] ? passwordValidationError[0].message: '';
        return (
            <div>                           
                <div className={styles.modal__header}>                    
                    <a onClick={this.props.baseComponent} className={styles.modal__back}><i className='fas fa-arrow-left'></i></a>
                    <span>Sign Up</span>
                </div>
                <hr className={styles.modal__divider}/>
                <form className={styles.modal__form} name='signUpForm' onSubmit={this.signUp} role='form' noValidate>
                    <label className={styles.modal__form__label}>Email<span className={styles.modal__form__label__required}>*</span></label>
                    <input className={emailValidationError === true || !this.state.email ? styles.modal__form__input : styles.modal__form__input__error} name='email' type='email' placeholder='Email' onChange={this.handleChange}/>
                    <div className={styles.modal__form__error}>
                        <span className={`${styles.modal__form__error__message} ${!this.state.email ? styles.modal__form__error__message__hidden : ''}`}>{emailErrorMessage}</span>
                    </div>
                    <label className={styles.modal__form__label}>Password<span className={styles.modal__form__label__required}>*</span></label>
                    <input className={passwordValidationError === true || !this.state.password ? styles.modal__form__input : styles.modal__form__input__error} name='password' type='password' placeholder='Password'onChange={this.handleChange}/>
                    <div className={styles.modal__form__error}>
                        <span className={`${styles.modal__form__error__message} ${!this.state.password ? styles.modal__form__error__message__hidden : ''}`}>{passwordErrorMessage}</span>
                    </div>
                    <label className={styles.modal__form__label}>Full Name</label>
                    <input className={styles.modal__form__input__small} name='firstName' type='text' placeholder='First' onChange={this.handleChange}/>&nbsp;
                    <input className={styles.modal__form__input__small} name='lastName' type='text' placeholder='Last' onChange={this.handleChange}/>
                    <div className={styles.modal__form__group}>
                    <input className={styles.modal__form__group__submit} disabled={emailValidationError !== true || passwordValidationError !== true} type='submit' value='Create Account'/>
                    </div>
                </form>
            </div>
        );
    }
}

SignUp.propTypes = {
    baseComponent: PropTypes.func.isRequired
};