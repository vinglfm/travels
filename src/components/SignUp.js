import React, {Component} from 'react';
import PropTypes from 'prop-types';
import styles from './SignInModal.css';

export default class SignUp extends Component {
    constructor(props) {
        super(props);

        this.signUp = this.signUp.bind(this);
    }

    signUp() {
        console.log(this.refs.email.value);
        console.log(this.refs.password.value);
    }

    render() {
        return (
            <div>           
                <div className={styles.modal__header}>
                    <a onClick={this.props.baseComponent} className={styles.modal__back}><i className='fas fa-arrow-left'></i></a>
                    <span>Sign Up</span>
                </div>
                <hr className={styles.modal__divider}/>
                <form className={styles.modal__form} name='signUpForm' onSubmit={this.signUp} role='form'>
                    <label className={styles.modal__form__label}>Email<span className={styles.modal__form__label__required}>*</span></label>
                    <input className={styles.modal__form__input} type='email' ref='email'/>
                    <label className={styles.modal__form__label}>Password<span className={styles.modal__form__label__required}>*</span></label>
                    <input className={styles.modal__form__input} type='password' ref='password' placeholder='Password' minLength='8' maxLength='16'/>
                    <label className={styles.modal__form__label}>Full Name</label>
                    <input className={styles.modal__form__input__small} type='text' ref='firstName' placeholder='First'/>&nbsp;
                    <input className={styles.modal__form__input__small} type='text' ref='lastName' placeholder='Last'/>
                    <div className={styles.modal__form__group}>
                    <input className={styles.modal__form__group__submit} type='submit' value='Create Account'/>
                    </div>
                </form>
            </div>
        );
    }
}

SignUp.propTypes = {
    baseComponent: PropTypes.func.isRequired
};