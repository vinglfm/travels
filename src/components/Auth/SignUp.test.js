import enzyme, {shallow} from 'enzyme';
import ReactSixteenAdapter from 'enzyme-adapter-react-16';
enzyme.configure({ adapter: new ReactSixteenAdapter() });

import React from 'react';
import {SignUp} from './SignUp';

describe('<SignUp />', () => {
  let wrapper;
  let onBack = jest.fn();
  let onClose = jest.fn();
  beforeEach(() => {
    wrapper = shallow(<SignUp onBack={onBack} onClose={onClose}/>);
  });

  it('renders Component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
