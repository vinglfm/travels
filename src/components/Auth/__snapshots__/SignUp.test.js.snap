// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp /> renders Component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUp
    onBack={[Function]}
    onClose={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <FormHeader
          onBack={[Function]}
          title="Sign Up"
/>,
        <form
          className={undefined}
          name="signUpForm"
          noValidate={true}
          onSubmit={[Function]}
>
          <TextInput
                    error={true}
                    errorMessage="Email is not valid"
                    label="Email"
                    name="email"
                    onChange={[Function]}
                    required="true"
                    type="email"
                    value=""
          />
          <TextInput
                    error={true}
                    errorMessage="Password should be between 5 and 12 characters"
                    label="Password"
                    name="password"
                    onChange={[Function]}
                    required="true"
                    type="password"
                    value=""
          />
          <label
                    className={undefined}
          >
                    Full Name
          </label>
          <input
                    className={undefined}
                    name="firstName"
                    onChange={[Function]}
                    placeholder="First"
                    type="text"
          />
           
          <input
                    className={undefined}
                    name="lastName"
                    onChange={[Function]}
                    placeholder="Last"
                    type="text"
          />
          <div
                    className={undefined}
          >
                    <input
                              className={undefined}
                              disabled={true}
                              type="submit"
                              value="Create Account"
                    />
          </div>
</form>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onBack": [Function],
          "title": "Sign Up",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <TextInput
              error={true}
              errorMessage="Email is not valid"
              label="Email"
              name="email"
              onChange={[Function]}
              required="true"
              type="email"
              value=""
/>,
            <TextInput
              error={true}
              errorMessage="Password should be between 5 and 12 characters"
              label="Password"
              name="password"
              onChange={[Function]}
              required="true"
              type="password"
              value=""
/>,
            <label
              className={undefined}
>
              Full Name
</label>,
            <input
              className={undefined}
              name="firstName"
              onChange={[Function]}
              placeholder="First"
              type="text"
/>,
            " ",
            <input
              className={undefined}
              name="lastName"
              onChange={[Function]}
              placeholder="Last"
              type="text"
/>,
            <div
              className={undefined}
>
              <input
                            className={undefined}
                            disabled={true}
                            type="submit"
                            value="Create Account"
              />
</div>,
          ],
          "className": undefined,
          "name": "signUpForm",
          "noValidate": true,
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "error": true,
              "errorMessage": "Email is not valid",
              "label": "Email",
              "name": "email",
              "onChange": [Function],
              "required": "true",
              "type": "email",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "error": true,
              "errorMessage": "Password should be between 5 and 12 characters",
              "label": "Password",
              "name": "password",
              "onChange": [Function],
              "required": "true",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Full Name",
              "className": undefined,
            },
            "ref": null,
            "rendered": "Full Name",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": undefined,
              "name": "firstName",
              "onChange": [Function],
              "placeholder": "First",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": undefined,
              "name": "lastName",
              "onChange": [Function],
              "placeholder": "Last",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <input
                className={undefined}
                disabled={true}
                type="submit"
                value="Create Account"
/>,
              "className": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": undefined,
                "disabled": true,
                "type": "submit",
                "value": "Create Account",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "type": "div",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <FormHeader
            onBack={[Function]}
            title="Sign Up"
/>,
          <form
            className={undefined}
            name="signUpForm"
            noValidate={true}
            onSubmit={[Function]}
>
            <TextInput
                        error={true}
                        errorMessage="Email is not valid"
                        label="Email"
                        name="email"
                        onChange={[Function]}
                        required="true"
                        type="email"
                        value=""
            />
            <TextInput
                        error={true}
                        errorMessage="Password should be between 5 and 12 characters"
                        label="Password"
                        name="password"
                        onChange={[Function]}
                        required="true"
                        type="password"
                        value=""
            />
            <label
                        className={undefined}
            >
                        Full Name
            </label>
            <input
                        className={undefined}
                        name="firstName"
                        onChange={[Function]}
                        placeholder="First"
                        type="text"
            />
             
            <input
                        className={undefined}
                        name="lastName"
                        onChange={[Function]}
                        placeholder="Last"
                        type="text"
            />
            <div
                        className={undefined}
            >
                        <input
                                    className={undefined}
                                    disabled={true}
                                    type="submit"
                                    value="Create Account"
                        />
            </div>
</form>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onBack": [Function],
            "title": "Sign Up",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <TextInput
                error={true}
                errorMessage="Email is not valid"
                label="Email"
                name="email"
                onChange={[Function]}
                required="true"
                type="email"
                value=""
/>,
              <TextInput
                error={true}
                errorMessage="Password should be between 5 and 12 characters"
                label="Password"
                name="password"
                onChange={[Function]}
                required="true"
                type="password"
                value=""
/>,
              <label
                className={undefined}
>
                Full Name
</label>,
              <input
                className={undefined}
                name="firstName"
                onChange={[Function]}
                placeholder="First"
                type="text"
/>,
              " ",
              <input
                className={undefined}
                name="lastName"
                onChange={[Function]}
                placeholder="Last"
                type="text"
/>,
              <div
                className={undefined}
>
                <input
                                className={undefined}
                                disabled={true}
                                type="submit"
                                value="Create Account"
                />
</div>,
            ],
            "className": undefined,
            "name": "signUpForm",
            "noValidate": true,
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "error": true,
                "errorMessage": "Email is not valid",
                "label": "Email",
                "name": "email",
                "onChange": [Function],
                "required": "true",
                "type": "email",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "error": true,
                "errorMessage": "Password should be between 5 and 12 characters",
                "label": "Password",
                "name": "password",
                "onChange": [Function],
                "required": "true",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Full Name",
                "className": undefined,
              },
              "ref": null,
              "rendered": "Full Name",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": undefined,
                "name": "firstName",
                "onChange": [Function],
                "placeholder": "First",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": undefined,
                "name": "lastName",
                "onChange": [Function],
                "placeholder": "Last",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <input
                  className={undefined}
                  disabled={true}
                  type="submit"
                  value="Create Account"
/>,
                "className": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": undefined,
                  "disabled": true,
                  "type": "submit",
                  "value": "Create Account",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              "type": "div",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
